scripts/openmc-update-mgxs

```diff
@@ -0,0 +1,225 @@
+                        help='input XML file')
+    parser.add_argument('-o', '--output', nargs='?', default='',
+                        help='output file, in HDF5 format')
+    args = vars(parser.parse_args())
+
+    if args['output'] == '':
+        filename = args['input'].name
+        extension = os.path.splitext(filename)
+        if extension == '.xml':
+            filename = filename[:filename.rfind('.')] + '.h5'
+        args['output'] = filename
+
+    # Parse and return commandline arguments.
+    return args
+
+
+def get_data(element, entry):
+    value = element.find(entry)
+    if value is not None:
+        value = value.text.strip()
+    else:
+        if entry in element.attrib:
+            value = element.attrib[entry].strip()
+        else:
+            value = None
+
+    return value
+
+
+if __name__ == '__main__':
+    args = parse_args()
+
+    # Parse the XML data.
+    tree = ET.parse(args['input'])
+    root = tree.getroot()
+
+    # Get old metadata
+    temp = tree.find('group_structure').text.strip()
+    temp = np.array(temp.split())
+    group_structure = temp.astype(np.float)
+    # Convert from MeV to eV
+    group_structure *= 1.e6
+    energy_groups = openmc.mgxs.EnergyGroups(group_structure)
+    temp = tree.find('inverse-velocity')
+    if temp is not None:
+        temp = temp.text.strip()
+        temp = np.array(temp.split())
+        inverse_velocity = temp.astype(np.float)
+    else:
+        inverse_velocity = None
+
+    xsd = []
+    names = []
+
+    # Now move on to the cross section data itself
+    for xsdata_elem in root.iter('xsdata'):
+        name = get_data(xsdata_elem, 'name')
+
+        temperature = get_data(xsdata_elem, 'kT')
+        if temperature is not None:
+            temperature = \
+                float(temperature) / openmc.data.K_BOLTZMANN * 1.E6
+        else:
+            temperature = 294.
+        temperatures = [temperature]
+
+        awr = get_data(xsdata_elem, 'awr')
+        if awr is not None:
+            awr = float(awr)
+
+        representation = get_data(xsdata_elem, 'representation')
+        if representation is None:
+            representation = 'isotropic'
+        if representation == 'angle':
+            n_azi = int(get_data(xsdata_elem, 'num_azimuthal'))
+            n_pol = int(get_data(xsdata_elem, 'num_polar'))
+
+        scatter_format = get_data(xsdata_elem, 'scatt_type')
+        if scatter_format is None:
+            scatter_format = 'legendre'
+
+        order = int(get_data(xsdata_elem, 'order'))
+
+        tab_leg = get_data(xsdata_elem, 'tabular_legendre')
+        if tab_leg is not None:
+            warnings.Warning('The tabular_legendre option has moved to the '
+                             'settings.xml file and must be added manually')
+
+        # Either add the data to a previously existing xsdata (if it is
+        # for the same 'name' but a different temperature), or create a
+        # new one.
+        try:
```

Comment:
This is very long method, consider breaking into smaller methods for better readability.